feature/google/src/main/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -349,29 +350,46 @@ class BillingWrapper(
     }
 
     class GoogleQueryPurchasesResult(
-        @BillingClient.BillingResponseCode responseCode: Int,
+        isSuccessful: Boolean,
         purchasesByHashedToken: Map<String, PurchaseWrapper>
-    ) : QueryPurchasesResult(responseCode, purchasesByHashedToken) {
+    ) : QueryPurchasesResult(isSuccessful, purchasesByHashedToken)
 
-        override fun isSuccessful(): Boolean = responseCode == BillingClient.BillingResponseCode.OK
-    }
+    private fun Purchase.PurchasesResult.isSuccessful() = responseCode == BillingClient.BillingResponseCode.OK
 
-    override fun queryPurchases(@SkuType skuType: String): QueryPurchasesResult? {
-        return billingClient?.let { billingClient ->
-            log(LogIntent.DEBUG, RestoreStrings.QUERYING_PURCHASE.format(skuType))
-            val result = billingClient.queryPurchases(skuType)
+    override fun queryPurchases(
+        appUserID: String,
+        completion: (QueryPurchasesResult) -> Unit
+    ) {
+        billingClient?.let { billingClient ->
+            log(LogIntent.DEBUG, RestoreStrings.QUERYING_PURCHASE)
 
-            val purchasesList = result.purchasesList ?: emptyList<Purchase>()
+            val queryActiveSubscriptionsResult = billingClient.queryPurchases(SkuType.SUBS)
+            val queryUnconsumedInAppsRequest = billingClient.queryPurchases(SkuType.INAPP)
 
-            GoogleQueryPurchasesResult(
-                result.responseCode,
-                purchasesList.map { purchase ->
-                    val hash = purchase.purchaseToken.sha1()
-                    log(LogIntent.DEBUG, RestoreStrings.QUERYING_PURCHASE_WITH_HASH.format(skuType, hash))
-                    hash to GooglePurchaseWrapper(purchase, skuType.toProductType(), presentedOfferingIdentifier = null)
-                }.toMap()
-            )
-        }
+            val isSuccessful =
+                queryActiveSubscriptionsResult.isSuccessful() && queryUnconsumedInAppsRequest.isSuccessful()
+
+            val activeSubscriptionsList = queryActiveSubscriptionsResult.purchasesList ?: emptyList<Purchase>()
+            val mapOfActiveSubscriptions = activeSubscriptionsList.toMapOfGooglePurchaseWrapper(SkuType.SUBS)
+
+            val unconsumedInAppsList = queryUnconsumedInAppsRequest.purchasesList ?: emptyList<Purchase>()
+            val mapOfUnconsumedInApps = unconsumedInAppsList.toMapOfGooglePurchaseWrapper(SkuType.INAPP)
+
+            completion(GoogleQueryPurchasesResult(
+                isSuccessful,
+                mapOfActiveSubscriptions + mapOfUnconsumedInApps
+            ))
+        } ?: completion(GoogleQueryPurchasesResult(isSuccessful = false, emptyMap()))
```

Comment:
is there a way to return an error here? was this case not handled previously?

Reply:
Yeah, this case wasn't handled before. With this change, `updatePendingPurchaseQueue` will now get a result with `isSuccessful` false. And `updatePendingPurchaseQueue` right now is not doing anything when `isSuccessful` is `false`, so nothing has really changed... 

I would like to remove the `QueryPurchasesResult` and change with an onCompletion and an onError, to be more consistent with the rest of the code. But for now, I am going to add a `PurchasesError` to the `GoogleQueryPurchasesResult` so that we can log it in `updatePendingPurchaseQueue`